src/test/basics/StringUtilities_test.cpp

```diff
@@ -63,23 +63,133 @@ class StringUtilities_test : public beast::unit_test::suite
     {
         testcase ("parseUrl");
 
-        parsedURL pUrl;
-
-        BEAST_EXPECT(parseUrl (pUrl, "lower://domain"));
-        BEAST_EXPECT(pUrl.scheme == "lower");
-        BEAST_EXPECT(pUrl.domain == "domain");
-        BEAST_EXPECT(! pUrl.port);
-        BEAST_EXPECT(pUrl.path == "");
-        BEAST_EXPECT(parseUrl (pUrl, "UPPER://domain:234/"));
-        BEAST_EXPECT(pUrl.scheme == "upper");
-        BEAST_EXPECT(*pUrl.port == 234);
-        BEAST_EXPECT(pUrl.path == "/");
-        BEAST_EXPECT(parseUrl (pUrl, "Mixed://domain/path"));
-        BEAST_EXPECT(pUrl.scheme == "mixed");
-        BEAST_EXPECT(pUrl.path == "/path");
-        BEAST_EXPECT(parseUrl (pUrl, "scheme://[::1]:123/path"));
-        BEAST_EXPECT(*pUrl.port == 123);
-        BEAST_EXPECT(pUrl.domain == "::1");
+        {
+            parsedURL pUrl;
+            BEAST_EXPECT(parseUrl (pUrl, "lower://domain"));
+            BEAST_EXPECT(pUrl.scheme == "lower");
+            BEAST_EXPECT(pUrl.username == "");
+            BEAST_EXPECT(pUrl.password == "");
```

Comment:
Instead of `== ""`, I think it's a bit clearer to use `.empty()`. eg. `BEAST_EXPECT(pUrl.username.empty());`